runtime/drivers/clickhouse/model_manager.go

```diff
@@ -0,0 +1,138 @@
+package clickhouse
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+type ModelInputProperties struct {
+	SQL string `mapstructure:"sql"`
+}
+
+func (p *ModelInputProperties) Validate() error {
+	if p.SQL == "" {
+		return fmt.Errorf("missing property 'sql'")
+	}
+	return nil
+}
```

Comment:
**Issue**: The validation method for 'ModelInputProperties' only checks if the SQL property is empty. Consider enhancing this validation to check for valid SQL syntax or against SQL injection risks. <br> **Fix**: Implement additional validation checks for SQL content, possibly using prepared statements or a library designed to safely handle SQL queries. <br> **Code Suggestion**: 
 ```
 - if p.SQL == ""
 + if !isValidSQL(p.SQL)
 ```